src/components/clusters/ClustersTable.tsx

```diff
@@ -38,20 +39,22 @@ const columns = [
 ];
 
 const ClustersTable: React.FC<ClustersTableProps> = ({ rows, deleteCluster }) => {
-  const [sortBy, setSortBy] = React.useState({
+  const [deleteClusterRow, setDeleteClusterRow] = React.useState<IRowData>();
+  const [sortBy, setSortBy] = React.useState<ISortBy>({
     index: 0, // Name-column
     direction: SortByDirection.asc,
-  } as ISortBy);
+  });
 
   const actions = React.useMemo(
     () => [
       {
         title: 'Delete',
-        onClick: (event: React.MouseEvent, rowIndex: number, rowData: IRowData) =>
-          deleteCluster(rowData.id.title),
+        onClick: (event: React.MouseEvent, rowIndex: number, rowData: IRowData) => {
+          setDeleteClusterRow(rowData);
```

Comment:
What is the benefit of storing whole row here? I would prefer just identification of the cluster to be deleted.

Reply:
Storing only name and id instead of the whole row now